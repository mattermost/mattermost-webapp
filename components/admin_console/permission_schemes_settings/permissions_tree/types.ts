// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type Permission = {
    id: string;
    combined?: boolean;
    permissions: string[];
}
export type Group = {
    id: string;
    permissions: Array<Permission | string>;
}

export type AdditionalValues = {
    edit_post: {
        editTimeLimitButton: JSX.Element;
    };
}
