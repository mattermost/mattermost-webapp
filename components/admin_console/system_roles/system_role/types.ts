// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type ReadAccess = 'read';
export const readAccess: ReadAccess = 'read';
export type WriteAccess = 'write';
export const writeAccess: WriteAccess = 'write';
export type NoAccess = false;
export const noAccess: NoAccess = false;
export type MixedAccess = 'mixed';
export const mixedAccess: MixedAccess = 'mixed';

export type PermissionAccess = ReadAccess | WriteAccess | NoAccess;
export type PermissionsToUpdate = Record<string, ReadAccess | WriteAccess | NoAccess>;
export type PermissionToUpdate = {
    name: string;
    value: PermissionAccess;
};

export type SystemSection = {
    name: string;
    hasDescription?: boolean;
    subsections?: SystemSection[];
    disabled?: boolean;
}
