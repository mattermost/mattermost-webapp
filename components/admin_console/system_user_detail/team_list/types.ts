// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Team, TeamMembership} from '@mattermost/types/teams';

export type TeamWithMembership = Team & TeamMembership;
