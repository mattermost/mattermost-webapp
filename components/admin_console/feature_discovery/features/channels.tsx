// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {t} from 'utils/i18n';

import FeatureDiscovery from '../index';

import ChannelsSVG from './images/channels_svg';

const ChannelsFeatureDiscovery: React.FC = () => {
    return (
        <FeatureDiscovery
            featureName='channels'
            titleID='admin.channels_feature_discovery.title'
            titleDefault='Create read-only channels with Mattermost Enterprise E20'
            copyID='admin.channels_feature_discovery.copy'
            copyDefault={'Decide which channels should be public, private, read-only, or react-only.'}
            learnMoreURL='https://docs.mattermost.com/deployment/team-channel-management.html'
            featureDiscoveryImage={<ChannelsSVG/>}
        />
    );
};

t('admin.channels_feature_discovery.title');
t('admin.channels_feature_discovery.copy');

export default ChannelsFeatureDiscovery;
