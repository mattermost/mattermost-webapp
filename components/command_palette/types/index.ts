// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export enum CommandPaletteEntities {
    Channel = 'channel',
    Playbooks = 'playbooks',
    Boards = 'boards',
    Messages = 'messages',
    Files = 'files',
    Runs = 'runs',
    GoTo = 'goTo',
}

export enum ChannelType {
    DM_CHANNEL= 'D',
    GM_CHANNEL= 'G',
    OPEN_CHANNEL= 'O',
    PRIVATE_CHANNEL= 'P',
}

export enum BoardsType {
    OPEN_CHANNEL= 'O',
    PRIVATE_CHANNEL= 'P',
}

export enum PlaybooksType {
    OPEN_CHANNEL= 'O',
    PRIVATE_CHANNEL= 'P',
}

