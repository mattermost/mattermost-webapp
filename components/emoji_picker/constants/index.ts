// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {EmojiCategory} from '@mattermost/types/emojis';

import * as Emoji from 'utils/emoji';

import {Category, Categories} from '../types';

export const RECENT = 'recent';
export const SEARCH_RESULTS = 'searchResults';
export const SMILEY_EMOTION = 'smileys-emotion';
export const CUSTOM = 'custom';

const categoryClass: Map<EmojiCategory, string> = new Map([
    [RECENT, 'icon-clock-outline'],
    [SMILEY_EMOTION, 'icon-emoticon-happy-outline'],
    ['people-body', 'icon-account-outline'],
    ['animals-nature', 'icon-leaf-outline'],
    ['food-drink', 'icon-food-apple'],
    ['activities', 'icon-basketball'],
    ['travel-places', 'icon-airplane-variant'],
    ['objects', 'icon-lightbulb-outline'],
    ['symbols', 'icon-heart-outline'],
    ['flags', 'icon-flag-outline'],
    [CUSTOM, 'icon-emoticon-custom-outline'],
    [SEARCH_RESULTS, ''],
]);

function createCategory(name: EmojiCategory): Category {
    return {
        name,
        id: Emoji.CategoryTranslations.get(name) || '',
        className: categoryClass.get(name) || '',
        message: Emoji.CategoryMessage.get(name)!,
    };
}

export const RECENT_EMOJI_CATEGORY: Pick<Categories, 'recent'> = {recent: createCategory(RECENT)};
export const SEARCH_EMOJI_CATEGORY: Pick<Categories, typeof SEARCH_RESULTS> = {searchResults: createCategory(SEARCH_RESULTS)};

export const CATEGORIES: Categories = Emoji.CategoryNames.
    filter((category) => !(category === 'recent' || category === 'searchResults')).
    reduce((previousCategory, currentCategory) => {
        return {
            ...previousCategory,
            [currentCategory]: createCategory(currentCategory as EmojiCategory),
        };
    }, {} as Categories);

export const EMOJI_PER_ROW = 9; // needs to match variable `$emoji-per-row` in _variables.scss
export const ITEM_HEIGHT = 36; //as per .emoji-picker__item height in _emoticons.scss
export const EMOJI_CONTAINER_HEIGHT = 290; // If this changes, the spaceRequiredAbove and spaceRequiredBelow props passed to the EmojiPickerOverlay must be updated

export const CATEGORY_HEADER_ROW = 'categoryHeaderRow';
export const EMOJIS_ROW = 'emojisRow';

export const EMOJI_SCROLL_THROTTLE_DELAY = 150;
export const EMOJI_ROWS_OVERSCAN_COUNT = 1;

export const CUSTOM_EMOJIS_PER_PAGE = 200;
export const CUSTOM_EMOJI_SEARCH_THROTTLE_TIME_MS = 1000;
