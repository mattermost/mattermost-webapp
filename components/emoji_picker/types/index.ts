// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {EmojiCategory, Emoji} from 'mattermost-redux/types/emojis';

export type Category = {
    className: string;
    emojiIds?: string[];
    id: string;
    message: string;
    name: EmojiCategory;
    offset: number;
};

export type Categories = Record<EmojiCategory, Category>;

export type CategoryOrEmojiRow = {
    type: 'categoryHeaderRow' | 'emojisRow';
    row: EmojiCategory | Emoji[];
}
