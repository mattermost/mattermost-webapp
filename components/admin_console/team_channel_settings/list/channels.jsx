// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {createSelector} from 'reselect';

import {getAllChannels as getData} from 'mattermost-redux/actions/channels';
import {getAllChannels} from 'mattermost-redux/selectors/entities/channels';

import {t} from 'utils/i18n';

import List from './channel_list.jsx';

const getSortedListOfChannels = createSelector(
    getAllChannels,
    (teams) => Object.values(teams).sort((a, b) => a.name.localeCompare(b.name))
);

function mapStateToProps(state) {
    return {
        data: getSortedListOfChannels(state),
        emptyListTextId: t('admin.team_settings.channel_list.no_channels_found'),
        emptyListTextDefaultMessage: 'No channels found',
    };
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            getData,
        }, dispatch),
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(List);
