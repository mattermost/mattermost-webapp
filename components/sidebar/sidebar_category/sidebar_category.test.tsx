// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {CategoryTypes} from 'mattermost-redux/constants/channel_categories';
import {ChannelType} from 'mattermost-redux/types/channels';

import SidebarCategory from 'components/sidebar/sidebar_category/sidebar_category';

describe('components/sidebar/sidebar_category', () => {
    const baseProps = {
        category: {
            id: 'category1',
            team_id: 'team1',
            type: CategoryTypes.CUSTOM,
            display_name: 'custom_category_1',
        },
        channels: [
            {
                id: 'channel_id',
                display_name: 'channel_display_name',
                create_at: 0,
                update_at: 0,
                delete_at: 0,
                team_id: '',
                type: 'O' as ChannelType,
                name: '',
                header: '',
                purpose: '',
                last_post_at: 0,
                total_msg_count: 0,
                extra_update_at: 0,
                creator_id: '',
                scheme_id: '',
                group_constrained: false,
            },
        ],
        setChannelRef: jest.fn(),
        getChannelRef: jest.fn(),
        handleOpenMoreDirectChannelsModal: jest.fn(),
        isCollapsed: false,
        actions: {
            setCategoryCollapsed: jest.fn(),
        },
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <SidebarCategory {...baseProps}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot when collapsed', () => {
        const props = {
            ...baseProps,
            isCollapsed: true,
        };

        const wrapper = shallow(
            <SidebarCategory {...props}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot when the category is DM and there are no DMs to display', () => {
        const props = {
            ...baseProps,
            category: {
                ...baseProps.category,
                type: CategoryTypes.DIRECT_MESSAGES,
            },
            channels: [],
        };

        const wrapper = shallow(
            <SidebarCategory {...props}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot when there are no channels to display', () => {
        const props = {
            ...baseProps,
            channels: [],
        };

        const wrapper = shallow(
            <SidebarCategory {...props}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should collapse the channel on toggle when it is not collapsed', () => {
        const wrapper = shallow<SidebarCategory>(
            <SidebarCategory {...baseProps}/>
        );

        wrapper.instance().handleCollapse();
        expect(baseProps.actions.setCategoryCollapsed).toHaveBeenCalledWith(baseProps.category.id, true);
    });

    test('should un-collapse the channel on toggle when it is collapsed', () => {
        const props = {
            ...baseProps,
            isCollapsed: true,
        };

        const wrapper = shallow<SidebarCategory>(
            <SidebarCategory {...props}/>
        );

        wrapper.instance().handleCollapse();
        expect(baseProps.actions.setCategoryCollapsed).toHaveBeenCalledWith(props.category.id, false);
    });
});
