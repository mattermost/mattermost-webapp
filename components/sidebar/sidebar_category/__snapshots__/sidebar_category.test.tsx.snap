// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/sidebar/sidebar_category should match snapshot 1`] = `
<div
  className="SidebarChannelGroup a11y__section"
>
  <div
    className="SidebarChannelGroupHeader"
  >
    <button
      aria-label="custom_category_1"
      className="SidebarChannelGroupHeader_groupButton"
      onClick={[Function]}
    >
      <i
        className="icon icon-chevron-down"
      />
      <div>
        custom_category_1
      </div>
    </button>
  </div>
  <div
    className="SidebarChannelGroup_content"
  >
    <ul
      className="NavGroupContent"
      role="list"
    >
      <Connect(SidebarChannel)
        channelId="channel_id"
        getChannelRef={[MockFunction]}
        isCategoryCollapsed={false}
        key="channel_id"
        setChannelRef={[MockFunction]}
      />
    </ul>
  </div>
</div>
`;

exports[`components/sidebar/sidebar_category should match snapshot when collapsed 1`] = `
<div
  className="SidebarChannelGroup a11y__section"
>
  <div
    className="SidebarChannelGroupHeader"
  >
    <button
      aria-label="custom_category_1"
      className="SidebarChannelGroupHeader_groupButton"
      onClick={[Function]}
    >
      <i
        className="icon icon-chevron-down icon-rotate-minus-90"
      />
      <div>
        custom_category_1
      </div>
    </button>
  </div>
  <div
    className="SidebarChannelGroup_content"
  >
    <ul
      className="NavGroupContent"
      role="list"
    >
      <Connect(SidebarChannel)
        channelId="channel_id"
        getChannelRef={[MockFunction]}
        isCategoryCollapsed={true}
        key="channel_id"
        setChannelRef={[MockFunction]}
      />
    </ul>
  </div>
</div>
`;

exports[`components/sidebar/sidebar_category should match snapshot when the category is DM and there are no DMs to display 1`] = `
<div
  className="SidebarChannelGroup a11y__section"
>
  <div
    className="SidebarChannelGroupHeader"
  >
    <button
      aria-label="custom_category_1"
      className="SidebarChannelGroupHeader_groupButton"
      onClick={[Function]}
    >
      <i
        className="icon icon-chevron-down hide-arrow"
      />
      <div>
        custom_category_1
      </div>
      <button
        aria-label="Create new direct message"
        className="SidebarChannelGroupHeader_addButton"
        onClick={[Function]}
      >
        <OverlayTrigger
          defaultOverlayShown={false}
          delayShow={500}
          overlay={
            <Tooltip
              bsClass="tooltip"
              className="hidden-xs"
              id="new-group-tooltip"
              placement="right"
            >
              Create new direct message
            </Tooltip>
          }
          placement="top"
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <i
            className="icon-plus"
          />
        </OverlayTrigger>
      </button>
    </button>
  </div>
  <div
    className="SidebarChannelGroup_content"
  >
    <ul
      className="NavGroupContent"
      role="list"
    />
  </div>
</div>
`;

exports[`components/sidebar/sidebar_category should match snapshot when there are no channels to display 1`] = `""`;
